### PR TITLE
fix an issue where o/p of show table extended was not correctly parsed for the metadata

### PR DESCRIPTION
**Pull request by tovganesh**

### CODE REVIEW

**Comment by tovganesh** [comment]
sample dict that this code is parsing:
<pre>
[{'name': '# col_name            ', 'type': 'data_type           ', 'comment': 'comment             '}, {'name': '', 'type': None, 'comment': None}, {'name': 'id', 'type': 'tinyint', 'comment': None}, {'name': '', 'type': None, 'comment': None}, {'name': '# Detailed Table Information', 'type': None, 'comment': None}, {'name': 'Database:           ', 'type': 'dbtdemo             ', 'comment': None}, {'name': 'Owner:              ', 'type': 'ganesh.venkateshwara', 'comment': None}, {'name': 'CreateTime:         ', 'type': 'Wed Aug 17 05:10:32 UTC 2022', 'comment': None}, {'name': 'LastAccessTime:     ', 'type': 'UNKNOWN             ', 'comment': None}, {'name': 'Protect Mode:       ', 'type': 'None                ', 'comment': None}, {'name': 'Retention:          ', 'type': '0                   ', 'comment': None}, {'name': 'Table Type:         ', 'type': 'VIRTUAL_VIEW        ', 'comment': None}, {'name': 'Table Parameters:', 'type': None, 'comment': None}, {'name': '', 'type': 'transient_lastDdlTime', 'comment': '1660713032          '}, {'name': '', 'type': None, 'comment': None}, {'name': '# Storage Information', 'type': None, 'comment': None}, {'name': 'SerDe Library:      ', 'type': '                    ', 'comment': None}, {'name': 'InputFormat:        ', 'type': '                    ', 'comment': None}, {'name': 'OutputFormat:       ', 'type': '                    ', 'comment': None}, {'name': 'Compressed:         ', 'type': 'No                  ', 'comment': None}, {'name': 'Num Buckets:        ', 'type': '0                   ', 'comment': None}, {'name': 'Bucket Columns:     ', 'type': '[]                  ', 'comment': None}, {'name': 'Sort Columns:       ', 'type': '[]                  ', 'comment': None}, {'name': '', 'type': None, 'comment': None}, {'name': '# View Information', 'type': None, 'comment': None}, {'name': 'View Original Text: ', 'type': 'SELECT * FROM dbtdemo.my_first_dbt_model WHERE id = 1', 'comment': None}, {'name': 'View Expanded Text: ', 'type': 'SELECT * FROM dbtdemo.my_first_dbt_model WHERE id = 1', 'comment': None}]
</pre>

File: dbt/adapters/impala/impl.py [hunk 1]
```diff
@@ -186,15 +186,17 @@ def parse_describe_extended(
...
         # Remove rows that start with a hash, they are comments
         rows = [
             row for row in raw_rows[0:pos]
             if not row['name'].startswith('#') and not row['name'] == ''
         ]
+        # trim the fields so that these are clean key,value pairs and metadata.get() correctly returns the values
         metadata = {
-            col['name']: col['type'] for col in raw_rows[pos + 1:]
+            col['name'].split(":")[0].strip(): col['type'].strip() for col in raw_rows[pos + 1:]
+            if not col['name'].startswith('#') and not col['name'] == '' and col['name'] and col['type']
```
Comment (sujitkp-blr) [inline dbt/adapters/impala/impl.py:199]: pls change this to check for col['name'] first and then conditions for startswith and ''

File: dbt/adapters/impala/impl.py [hunk 2]
```diff
@@ -186,15 +186,17 @@ def parse_describe_extended(
...
 
         # Remove rows that start with a hash, they are comments
         rows = [
             row for row in raw_rows[0:pos]
             if not row['name'].startswith('#') and not row['name'] == ''
         ]
+        # trim the fields so that these are clean key,value pairs and metadata.get() correctly returns the values
         metadata = {
-            col['name']: col['type'] for col in raw_rows[pos + 1:]
+            col['name'].split(":")[0].strip(): col['type'].strip() for col in raw_rows[pos + 1:]
```
Comment (sujitkp-blr) [inline dbt/adapters/impala/impl.py:198]: can u give a sample of col['name'] in the comments to understand how it is being split

Comment (tovganesh) [reply dbt/adapters/impala/impl.py:198]: sample is already added in above comment

File: dbt/adapters/impala/impl.py (hunk 1 above)
Comment (tovganesh) [reply dbt/adapters/impala/impl.py]: thanks, updated the code.

**Review by sujitkp-blr — APPROVED** [review]

**Pull request merged**